modules/oe_webtools_etrans/src/Plugin/Block/UnifiedEtransBlock.php

```diff
@@ -0,0 +1,309 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\oe_webtools_etrans\Plugin\Block;
+
+use Drupal\Component\Serialization\Json;
+use Drupal\Component\Utility\Html;
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Block\Attribute\Block;
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Language\LanguageInterface;
+use Drupal\Core\Language\LanguageManagerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\Core\Url;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a block that unifies Drupal-Etrans translation of current page.
+ */
+#[Block(
+  id: 'oe_webtools_etrans_unified',
+  admin_label: new TranslatableMarkup('OpenEuropa Unified eTrans'),
+  category: new TranslatableMarkup('Webtools')
+)]
+class UnifiedEtransBlock extends BlockBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * Default values for the configuration options of this block.
+   */
+  protected const DEFAULT_CONFIGURATION = [
+    'delay' => 0,
+    'include' => '',
+    'exclude' => '',
+  ];
+
+  /**
+   * Creates an ETransBlock.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Language\LanguageManagerInterface $languageManager
+   *   The language manager.
+   * @param \Drupal\Core\Routing\RouteMatchInterface $routeMatch
+   *   The route match service.
+   */
+  public function __construct(
+    array $configuration,
+    $plugin_id,
+    $plugin_definition,
+    protected LanguageManagerInterface $languageManager,
+    protected RouteMatchInterface $routeMatch,
+  ) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition): self {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('language_manager'),
+      $container->get('current_route_match')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function access(AccountInterface $account, $return_as_object = FALSE) {
+    $translation_exists = (bool) $this->getRouteEntityLangcode();
+    return AccessResult::allowedIf(!$translation_exists);
```

Comment:
We should apply route cache context.

Reply:
Please provide more info or doc, I do not understand.

Reply:
Recently I learned that checkAccess gets called regardless, so it doesn't really matter.